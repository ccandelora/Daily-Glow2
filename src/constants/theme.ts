export const COLORS = {
  // Primary colors for emotions (keeping these for emotional context)
  primary: {
    red: '#FF6B6B',     // Warm red for negative emotions
    green: '#00C853',   // Vibrant green for positive emotions
    blue: '#2979FF',    // Bright blue for calm emotions
    yellow: '#FFD600',  // Bright yellow for energetic emotions
  },
  
  // UI colors updated for dark theme with glowing accents
  ui: {
    background: 'rgba(28, 14, 45, 0.95)',  // Dark purple background
    card: 'rgba(38, 20, 60, 0.85)',        // Slightly lighter purple for cards
    text: '#FFFFFF',                        // White text for contrast
    textSecondary: 'rgba(255, 255, 255, 0.7)', // Semi-transparent white
    border: 'rgba(82, 67, 194, 0.3)',      // Subtle purple border
    disabled: 'rgba(82, 67, 194, 0.2)',    // Dimmed purple for disabled state
    accent: '#4169E1',                     // Royal blue for accents
    highlight: 'rgba(65, 105, 225, 0.15)', // Soft blue highlight
  },

  // Status colors with glow effects
  status: {
    success: '#00E676',
    error: '#FF5252',
    warning: '#FFD600',
    info: '#4169E1',
  },

  // Gradient colors for glow effects
  gradient: {
    start: 'rgba(65, 105, 225, 0.2)',  // Royal blue glow
    middle: 'rgba(147, 112, 219, 0.15)', // Purple glow
    end: 'rgba(28, 14, 45, 0.9)',     // Dark purple base
  },
};

export const FONTS = {
  sizes: {
    xs: 12,
    sm: 14,
    md: 16,
    lg: 22,
    xl: 28,
    xxl: 36,
    xxxl: 44,
  },
  weights: {
    regular: '400' as const,
    medium: '500' as const,
    semibold: '600' as const,
    bold: '700' as const,
  },
  // Adding font families for more sophistication
  families: {
    heading: 'System',
    body: 'System',
  }
};

export const SPACING = {
  xs: 4,
  sm: 8,
  md: 16,
  lg: 24,
  xl: 32,
  xxl: 40,
};

export const BORDER_RADIUS = {
  sm: 4,
  md: 8,
  lg: 16,
  xl: 24,
  circle: 999,
};

export const TIME_PERIODS = {
  MORNING: {
    label: 'Morning',
    range: { start: 5, end: 11 }, // 5:00 AM - 11:59 AM
    greeting: 'Good morning',
    icon: 'üåÖ'
  },
  AFTERNOON: {
    label: 'Afternoon',
    range: { start: 12, end: 16 }, // 12:00 PM - 4:59 PM
    greeting: 'Good afternoon',
    icon: '‚òÄÔ∏è'
  },
  EVENING: {
    label: 'Evening',
    range: { start: 17, end: 4 }, // 5:00 PM - 4:59 AM
    greeting: 'Good evening',
    icon: 'ÔøΩÔøΩ'
  }
} as const;

export default {
  COLORS,
  FONTS,
  SPACING,
  BORDER_RADIUS,
  TIME_PERIODS,
}; 