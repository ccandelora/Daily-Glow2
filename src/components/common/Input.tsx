import React from 'react';
import { TextInput, View, StyleSheet, TextInputProps, ViewStyle } from 'react-native';
import { Typography } from './Typography';
import theme from '../../constants/theme';

interface InputProps extends TextInputProps {
  label?: string;
  error?: string;
  containerStyle?: ViewStyle;
  multiline?: boolean;
}

export const Input: React.FC<InputProps> = ({
  label,
  error,
  containerStyle,
  style,
  multiline = false,
  ...props
}) => {
  return (
    <View style={[styles.container, containerStyle]}>
      {label && (
        <Typography variant="caption" style={styles.label}>
          {label}
        </Typography>
      )}
      <TextInput
        style={[
          styles.input,
          multiline && styles.multiline,
          error && styles.errorInput,
          style,
        ]}
        placeholderTextColor={theme.COLORS.ui.textSecondary}
        multiline={multiline}
        {...props}
      />
      {error && (
        <Typography variant="caption" color={theme.COLORS.status.error} style={styles.error}>
          {error}
        </Typography>
      )}
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    marginBottom: theme.SPACING.md,
  },
  label: {
    marginBottom: theme.SPACING.xs,
  },
  input: {
    backgroundColor: theme.COLORS.ui.card,
    borderRadius: theme.BORDER_RADIUS.md,
    padding: theme.SPACING.md,
    color: theme.COLORS.ui.text,
    fontSize: theme.FONTS.sizes.md,
    borderWidth: 1,
    borderColor: theme.COLORS.ui.border,
  },
  multiline: {
    minHeight: 100,
    textAlignVertical: 'top',
  },
  errorInput: {
    borderColor: theme.COLORS.status.error,
  },
  error: {
    marginTop: theme.SPACING.xs,
  },
}); 