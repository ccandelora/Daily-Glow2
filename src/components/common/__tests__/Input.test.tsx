import React from 'react';
import { render, fireEvent } from '@testing-library/react-native';
import { ViewStyle } from 'react-native';
import { Input } from '../Input';
import theme from '@/constants/theme';

describe('Input Component', () => {
  test('renders correctly with default props', () => {
    const { getByTestId, toJSON } = render(
      <Input testID="test-input" placeholder="Enter text" />
    );
    
    expect(getByTestId('test-input')).toBeTruthy();
    expect(toJSON()).toMatchSnapshot('default-input');
  });
  
  test('displays label when provided', () => {
    const labelText = 'Username';
    const { getByText } = render(
      <Input label={labelText} placeholder="Enter username" />
    );
    
    expect(getByText(labelText)).toBeTruthy();
  });
  
  test('displays error message when provided', () => {
    const errorMessage = 'This field is required';
    const { getByText } = render(
      <Input error={errorMessage} placeholder="Enter text" />
    );
    
    const errorElement = getByText(errorMessage);
    expect(errorElement).toBeTruthy();
    
    // Check error text color
    expect(errorElement.props.style.some((style: any) => 
      style && style.color === theme.COLORS.status.error
    )).toBe(true);
  });
  
  test('applies error styles to input when error is provided', () => {
    const { getByTestId } = render(
      <Input 
        testID="error-input" 
        error="Error message" 
        placeholder="Enter text" 
      />
    );
    
    const inputElement = getByTestId('error-input');
    
    // Check that error style is applied (borderColor should be error color)
    expect(inputElement.props.style.some((style: any) => 
      style && style.borderColor === theme.COLORS.status.error
    )).toBe(true);
  });
  
  test('applies multiline styles when multiline prop is true', () => {
    const { getByTestId } = render(
      <Input 
        testID="multiline-input" 
        multiline={true} 
        placeholder="Enter text" 
      />
    );
    
    const inputElement = getByTestId('multiline-input');
    
    // Check that multiline is passed to TextInput
    expect(inputElement.props.multiline).toBe(true);
    
    // Check that multiline style is applied
    expect(inputElement.props.style.some((style: any) => 
      style && style.minHeight === 100 && style.textAlignVertical === 'top'
    )).toBe(true);
  });
  
  test('applies custom container styles when provided', () => {
    const containerStyle = { marginTop: 20, width: '90%' as any };
    const { toJSON } = render(
      <Input 
        containerStyle={containerStyle} 
        placeholder="Enter text" 
      />
    );
    
    // Using snapshot to verify container style since it's not directly accessible
    expect(toJSON()).toMatchSnapshot('input-with-container-style');
  });
  
  test('applies custom input styles when provided', () => {
    const inputStyle = { backgroundColor: '#F3F3F3', fontSize: 18 };
    const { getByTestId } = render(
      <Input 
        testID="styled-input" 
        style={inputStyle} 
        placeholder="Enter text" 
      />
    );
    
    const inputElement = getByTestId('styled-input');
    
    // Check that custom style is applied
    const styleArray = inputElement.props.style;
    expect(styleArray[styleArray.length - 1]).toMatchObject(inputStyle);
  });
  
  test('handles text input correctly', () => {
    const onChangeTextMock = jest.fn();
    const { getByTestId } = render(
      <Input 
        testID="change-input" 
        onChangeText={onChangeTextMock} 
        placeholder="Enter text" 
      />
    );
    
    const inputElement = getByTestId('change-input');
    const testText = 'Hello, world!';
    
    fireEvent.changeText(inputElement, testText);
    
    expect(onChangeTextMock).toHaveBeenCalledWith(testText);
  });
  
  test('passes additional props to TextInput', () => {
    const { getByTestId } = render(
      <Input 
        testID="props-input" 
        placeholder="Enter text"
        maxLength={20}
        autoCapitalize="none"
        secureTextEntry={true}
      />
    );
    
    const inputElement = getByTestId('props-input');
    
    expect(inputElement.props.maxLength).toBe(20);
    expect(inputElement.props.autoCapitalize).toBe('none');
    expect(inputElement.props.secureTextEntry).toBe(true);
  });
}); 