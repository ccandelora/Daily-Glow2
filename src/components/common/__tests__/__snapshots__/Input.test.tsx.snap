// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input Component applies custom container styles when provided: input-with-container-style 1`] = `
<View
  style={
    [
      {
        "marginBottom": 16,
      },
      {
        "marginTop": 20,
        "width": "90%",
      },
    ]
  }
>
  <TextInput
    multiline={false}
    placeholder="Enter text"
    placeholderTextColor="rgba(255, 255, 255, 0.7)"
    style={
      [
        {
          "backgroundColor": "rgba(38, 20, 60, 0.85)",
          "borderColor": "rgba(82, 67, 194, 0.3)",
          "borderRadius": 8,
          "borderWidth": 1,
          "color": "#FFFFFF",
          "fontSize": 16,
          "padding": 16,
        },
        false,
        undefined,
        undefined,
      ]
    }
  />
</View>
`;

exports[`Input Component renders correctly with default props: default-input 1`] = `
<View
  style={
    [
      {
        "marginBottom": 16,
      },
      undefined,
    ]
  }
>
  <TextInput
    multiline={false}
    placeholder="Enter text"
    placeholderTextColor="rgba(255, 255, 255, 0.7)"
    style={
      [
        {
          "backgroundColor": "rgba(38, 20, 60, 0.85)",
          "borderColor": "rgba(82, 67, 194, 0.3)",
          "borderRadius": 8,
          "borderWidth": 1,
          "color": "#FFFFFF",
          "fontSize": 16,
          "padding": 16,
        },
        false,
        undefined,
        undefined,
      ]
    }
    testID="test-input"
  />
</View>
`;
